uom_crm_dna.py

```diff
@@ -1,83 +1,235 @@
-import pandas as pd
+"""
+Script to see whether or not variants supplied by known reference standards are called for the CRM assay
+during uncertainty of measurement runs
+"""
+
+
 from glob import glob
 import subprocess
+import sigfig
+import argparse
+from datetime import datetime
+from datetime import date
+import yaml
+import csv
+
+
+###########################
+###	Functions	###
+###########################
+
+def parse_crm_samples(crm_config_list):
+        """
+        Load uncertainty of measurement sample IDs and their corresponding reference standard into a dictionary
+        """
+	# This is the uncertainty of measurment sample ID:reference standard files
+        config_file = open(sample_type, 'r')
+ 
+        config_dict = {}
+
+        for config in crm_config_list:
+
+        		# Loading in the yaml file
+                	sampleid_dict = yaml.load(config_file, Loader=yaml.FullLoader)
+        	
+        		# Convert contents of yaml file into a dictionary
+        		for key in sampleid_dict:
+               		
+               			config_dict[key] = sampleid_dict[key]
+                        
+        return config_dict
+
+
+
+def crm_sample_file_list(reference_dict, sample_path): 
+	"""
+	Get all UOM crm sample filepaths from last year into list
+	"""
+	
+	reference_dict = parse_crm_samples([crm_config_list])
```

Comment:
This is redundant as `reference_dict` is passed in as an argument on line 44, this way you're running the `parse_crm_samples` function twice, first as itself on line 221 and then from within this function on line 224. You've got the output saved to a variable on line 221 so pass into this function and you dont need to rerun.

Also there's two arguments required by this function on line 44 but only one passed into it on line 224